scene/2d/gpu_particles_2d.cpp

```diff
@@ -452,7 +457,7 @@ void GPUParticles2D::_notification(int p_what) {
 		RS::get_singleton()->canvas_item_add_particles(get_canvas_item(), particles, texture_rid);
 
 #ifdef TOOLS_ENABLED
-		if (Engine::get_singleton()->is_editor_hint() && (this == get_tree()->get_edited_scene_root() || get_tree()->get_edited_scene_root()->is_ancestor_of(this))) {
+		if (Engine::get_singleton()->is_editor_hint() && show_visibility_rect) {
```

Comment:
The property is only set inside editor, so the `Engine::get_singleton()->is_editor_hint()` part is no longer necessary.

Reply:
thanks!